contracts/swap/KyberDmm.sol

```diff
@@ -61,6 +80,45 @@ contract KyberDmm is BaseSwap {
         srcAmount = amounts[0];
     }
 
+    function getExpectedInWithImpact(GetExpectedInParams calldata params)
+        external
+        view
+        override
+        onlyProxyContract
+        returns (uint256 srcAmount, uint256 priceImpact)
+    {
+        address[] memory pools = parseExtraArgs(params.tradePath.length - 1, params.extraArgs);
+        IERC20[] memory tradePathErc = new IERC20[](params.tradePath.length);
+        for (uint256 i = 0; i < params.tradePath.length; i++) {
+            tradePathErc[i] = IERC20(params.tradePath[i]);
+        }
+        uint256[] memory amounts = dmmRouter.getAmountsIn(params.destAmount, pools, tradePathErc);
+        srcAmount = amounts[0];
+        priceImpact = getPriceImpact(srcAmount, params.destAmount, tradePathErc, pools);
+    }
+
+    function getPriceImpact(
+        uint256 srcAmount,
+        uint256 destAmount,
+        IERC20[] memory tradePathErc,
+        address[] memory pools
+    ) private view returns (uint256 priceImpact) {
+        uint256 quote = srcAmount;
+        for (uint256 i; i < pools.length; i++) {
+            IDMMPool pool = IDMMPool(pools[i]);
+            (uint256 reserveIn, uint256 reserveOut) = pool.getReserves();
+            if (tradePathErc[i] == pool.token1()) {
+                (reserveIn, reserveOut) = (reserveOut, reserveIn);
+            }
+            quote = UniswapV2Library.quote(quote, reserveIn, reserveOut);
+        }
+        if (quote <= destAmount) {
+            priceImpact = 0;
```

Comment:
Should we return a negative price impact here?

Reply:
Maybe, I don't even think this case will happen. Because of how the pool work, you just can't make a negative `priceImpact` :)